src/lib/pipeline/FetchUnit.cc

```diff
@@ -176,8 +176,16 @@ void FetchUnit::tick() {
       }
     } else if (loopBufferState_ == LoopBufferState::WAITING &&
                pc_ == loopBoundaryAddress_) {
-      // Once set loopBoundaryAddress_ is fetched, start to fill loop buffer
-      loopBufferState_ = LoopBufferState::FILLING;
+      // Now that the loopBoundaryAddress_ has been set, start filling Loop
+      // Buffer if the branch predictor tells us to reenter the detected loop
+      if (macroOp[0]->isBranch() && !macroOp[0]->getBranchPrediction().taken) {
+        // If branch is not taken then detected loop is no longer relevant,
+        // therefore Loop Buffer stays idle
+        loopBufferState_ = LoopBufferState::IDLE;
```

Comment:
"If branch is not taken then we aren't re-entering the detected loop, therefore Loop Buffer stays idle" for clarity